### PR TITLE
Add support for restore functionality from backup

### PR DESCRIPTION
**Pull request by svt-hchaudhari**
Signed-off-by: Hemant Chaudhari <hchaudhari@hpe.com>

### Description
Add support for the restore post endpoint that will enable the consumer to restore the original virtual machine or to create the new virtual machine

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass(`$ tox`).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [x] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: CHANGELOG.md
Comment (svt-kmcdaniel) [inline CHANGELOG.md]: Can you please move this up with all the other backup commands.  

We should try to keep the unreleased section cluster together to make release noting easier when the next version of the python package is released

File: examples/backups.py
Comment (svt-kmcdaniel) [inline examples/backups.py]: Spelling error

s/filers/filters/

File: examples/backups.py
Comment (svt-kmcdaniel) [inline examples/backups.py]: Preferred that these blank lines be removed?

the end statement belongs to the while loop

File: examples/backups.py
Comment (svt-kmcdaniel) [inline examples/backups.py:92]: Historically, the example script prints a header before the section of code.

I.e create_backup

File: examples/backups.py
Comment (svt-kmcdaniel) [inline examples/backups.py]: how does this output look without the double "\n\n"?

we kept the double "\n\n" in the datastore example output.  Should we keep the output consistent

File: examples/backups.py
Comment (svt-kmcdaniel) [inline examples/backups.py]: I think we should be careful about restoring a virtual machine on a customer's machine that may be running this example.  

We have to ensure the "do no harm" philosophy when writing examples

File: examples/backups.py
Comment (svt-kmcdaniel) [inline examples/backups.py]: you are creating a 1 TB datastore but never removing it

File: examples/backups.py
Comment (svt-kmcdaniel) [inline examples/backups.py]: nit, you have `Object` capitalize here, but `name` lower case in the following statement

File: simplivity/resources/backups.py
Comment (svt-kmcdaniel) [inline simplivity/resources/backups.py:177]: This can be rewritten with just one if statement.

```if datastore and isinstance(datastore, datastores.Datastore):```

File: tests/unit/resources/test_backups.py
Comment (svt-kmcdaniel) [inline tests/unit/resources/test_backups.py]: if true is set, "vm1" and None are not needed and are ignored in the call?

File: tests/unit/resources/test_backups.py
Comment (svt-kmcdaniel) [inline tests/unit/resources/test_backups.py]: could we move this right above the assert_called_once_with statement since it is not used anywhere else, or put the hash into that statement?

I think it will be easier to troubleshoot a failing unit test that way.

File: CHANGELOG.md
Comment (svt-hchaudhari) [reply CHANGELOG.md]: Agreed and updated the file accordingly

File: examples/backups.py
Comment (svt-hchaudhari) [reply examples/backups.py]: Updated

File: examples/backups.py
Comment (svt-hchaudhari) [reply examples/backups.py]: updated

File: examples/backups.py
Comment (svt-hchaudhari) [reply examples/backups.py:92]: Added the print statement

File: examples/backups.py
Comment (svt-hchaudhari) [reply examples/backups.py]: Updated

File: examples/backups.py
Comment (svt-hchaudhari) [reply examples/backups.py]: Updated

File: simplivity/resources/backups.py
Comment (svt-hchaudhari) [reply simplivity/resources/backups.py:177]: As discussed we cannot really change this logic as datastore can be name/object, also it is an optional parameter.

File: tests/unit/resources/test_backups.py
Comment (svt-hchaudhari) [reply tests/unit/resources/test_backups.py]: Updated

File: tests/unit/resources/test_backups.py
Comment (svt-hchaudhari) [reply tests/unit/resources/test_backups.py]: Updated

File: examples/backups.py
Comment (svt-hchaudhari) [reply examples/backups.py]: We agree with this, however in order to have this "do no harm" policy we would need following changes, 
Ask user to provide virtual machine name in so that we can get the VM by that name and perform the backup and restore. 
Also this being sample example it is very unlikely that it would be executed on Production system so that way it would not be doing any harm to the system.

@sijeesh Please let us know your views on it and way to move forward.

File: examples/backups.py
Comment (svt-hchaudhari) [reply examples/backups.py]: In order to remove the datastore we need to remove the VM that we have restored on it first. However as of today we do not have VM delete functionality coded yet.  

@sijeesh Please let us know , how to take this forward so shall we document the behavior till the time we get VM delete functionality.

Comment (svt-kmcdaniel) [reply examples/backups.py]: @svt-hchaudhari did you look at how it is done inside the virtual_machine example?

Comment (svt-hchaudhari) [reply examples/backups.py]: Agreed, updated the code

File: examples/backups.py
Comment (svt-hchaudhari) [reply examples/backups.py]: As per the current check-in VM and datatsore name is in variable now the changes are 'do no harm' resolving the conversation

**Review by sijeesh — APPROVED** [review]
LGTM

**Comment by sijeesh** [comment]
@svt-kmcdaniel, waiting for your approval.

**Review by svt-kmcdaniel — APPROVED** [review]

**Pull request merged**
